bolero-kani/src/lib.rs

```diff
@@ -143,6 +143,19 @@ pub mod lib {
         fn gen_bool(&mut self, _probability: Option<f32>) -> Option<bool> {
             Some(kani::any())
         }
+
+        fn gen_from_bytes<Gen, T>(
+            &mut self,
+            _len: std::ops::RangeInclusive<usize>,
+            mut gen: Gen,
+        ) -> Option<T>
+        where
+            Gen: FnMut(&[u8]) -> Option<(usize, T)>,
+        {
+            let value = gen(kani::any()).map(|v| v.1);
+            kani::assume(value.is_some());
+            value
+        }
```

Comment:
I don't think this will work with `kani`, since it doesn't implement `any` for `&[u8]` (I'm actually surprised it compiled - I will need to look into that separately).

Let's do this instead:

```suggestion
        fn gen_from_bytes<Gen, T>(
            &mut self,
            len: std::ops::RangeInclusive<usize>,
            mut gen: Gen,
        ) -> Option<T>
        where
            Gen: FnMut(&[u8]) -> Option<(usize, T)>,
        {
            let bytes = kani::vec::any_vec::<u8, 256>();
            kani::assume(len.contains(&bytes.len()));
            let value = gen(&bytes).map(|v| v.1);
            kani::assume(value.is_some());
            value
        }
```

Reply:
It's probably due to the `#[cfg(not(kani))]` stubs at the top of the file. TBH I have literally no idea how to use kani, so I wrote something that seemed to work and tests passed :sweat_smile: I guess the CI also runs without kani enabled?

Anyway, I've updated the code to your version and just added a stub above so that tests pass :)